blocks/hooks/custom-anchor-styles.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getBlockRandomAnchor, hasBlockSupport, getBlockType } from '../api';
+
+const CUSTOM_STYLE_ATTRIBUTES = [ 'backgroundColor', 'textColor' ];
+
+class AddAnchorWhenNeeded extends Component {
+	componentWillReceiveProps( nextProps ) {
+		if ( ! nextProps.attributes.anchor && CUSTOM_STYLE_ATTRIBUTES.some( attr => nextProps.attributes[ attr ] ) ) {
```

Comment:
Minor: Personally I find the test within the condition to be a bit hard to read (and make the line long), vs. assigning to a separate variable e.g. `hasCustomStyleAttribute = ...`

Reply:
Yes, extracting a variable which explains the condition is always a good idea 👍